tinygrad/codegen/linearizer.py

```diff
@@ -94,12 +94,12 @@ def __init__(self, ast:LazyOp, output_buffer:LazyBuffer):
     self.ast = ast.src[0] if ast.op == MovementOps.RESHAPE else ast
 
     # get the output buffers
-    self.bufs = [output_buffer] + dedup(get_buffers(ast))
+    self.bufs = [output_buffer] + dedup(ast.buffers)
 
     # key for lookup in cache (can change, str might not be right)
     # bufs are needed because kernels like f(x) = x + x and f(x, y) = x + y have the same str(ast), but are different kernels.
     # mapping the buffers to integers is required because a-b != b-a (and how would you tell a and b apart?)
-    self.key = f"ASTKernelKey ast={str(map_buffers({x:i for i,x in enumerate(self.bufs)}, ast))} bufs={self.bufs}"
+    self.key = (ast.map_buffers({x:i for i,x in enumerate(self.bufs)}).key, tuple(map(attrgetter("key"), self.bufs)))
```

Comment:
[x["key"] for x in self.bufs] is slower?

Reply:
Oh yeah this should just be `x.key for x in self.bufs`, there was a reason why i did that when testing but i can't remember it now